src/lib/branchpredictors/GenericPredictor.cc

```diff
@@ -115,6 +94,64 @@ void GenericPredictor::flush(uint64_t address) {
     }
     rasHistory_.erase(it);
   }
+
+  assert((ftq_.size() > 0) &&
+         "Cannot flush instruction from Branch Predictor "
+         "when the ftq is empty");
+  ftq_.pop_back();
```

Comment:
Just trying to think of a valid scenario where a flush occurs but the FTQ could be empty... Perhaps if there is a _very_ early load-store conflict that leads to a pipeline flush then this assert would be triggered erroneously? As such should this be an `if` surrounding the `pop_back` instead?

Reply:
I think that this shouldn't happen at all -- it would only happen if we have done something to the pipeline in SimEng that means there are too many flushed happening.  Therefore, I think an assert makes more sense.  It gets rid of a comparison in Release mode, and is there only to help us catch a bug introduced in future development.  However, happy to change if the team thinks an 'if' is better.

Reply:
Given you're explanation above, I'm happy to leave as an assert